src/main/java/macrobase/analysis/pipeline/BasicContextualBatchedPipeline.java

```diff
@@ -0,0 +1,128 @@
+package macrobase.analysis.pipeline;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Lists;
+
+import macrobase.analysis.classify.OutlierClassifier;
+import macrobase.analysis.contextualoutlier.Context;
+import macrobase.analysis.contextualoutlier.ContextualOutlierDetector;
+import macrobase.analysis.result.AnalysisResult;
+import macrobase.analysis.result.OutlierClassificationResult;
+import macrobase.analysis.summary.BatchSummarizer;
+import macrobase.analysis.summary.Summary;
+import macrobase.conf.ConfigurationException;
+import macrobase.conf.MacroBaseConf;
+import macrobase.datamodel.Datum;
+import macrobase.ingest.DataIngester;
+import macrobase.util.IterUtils;
+
+public class BasicContextualBatchedPipeline extends AbstractPipeline {
+
+    private static final Logger log = LoggerFactory.getLogger(BasicContextualBatchedPipeline.class);
+
+	public BasicContextualBatchedPipeline(MacroBaseConf conf) throws ConfigurationException {
+		super(conf);
+		// TODO Auto-generated constructor stub
+	}
+
+	
+	private List<AnalysisResult> allARs;
+	private int index = 0;
+	
+	private void run() throws ConfigurationException, SQLException, IOException{
+		allARs = new ArrayList<AnalysisResult>();
+		
+		//here we invoke contextual outlier detection
```

Comment:
Put another way, if we run BasicContextualBatchedPipeline, shouldn't it be implicit that we want to find contexts?